core/sql/langman/LmRoutineJavaObj.cpp

```diff
@@ -192,6 +192,7 @@ LmResult LmRoutineJavaObj::invokeRoutineMethod(
           (*emitRowPtr_)(NULL,0,&qs);
         }
     }
+  jni->DeleteLocalRef(jniResult);
```

Comment:
@DaveBirdsall, I think the problem is that the text quoted by Suresh is the only place where it talks about local references. Therefore we are left to guess what happens in the case where there is no such return.